### DISCUSSION TITLE
How to handle deprecation between major versions?

### DISCUSSION THREAD

**Discussion by kvistgaard** [eprocurementontology/eprocurementontology#619] [General]
It seems that [:isWithinFrameworkAgreement](https://github.com/OP-TED/ePO/blob/98b154c9a78256014a37f5328caaf3e8f9a0361f/implementation/ePO/owl_ontology/ePO_owl_core.ttl#L5691) is deprecated in the current version of ePO, but instead of declared as such, it is simply removed which has a negative impact on backward compatibility and traceability.

**Comment by cristianvasquez** [comment]
Hello,

The Ontology is being versioned according to [Semantic Versioning rules](https://docs.ted.europa.eu/epo-home/versioning.html), with major, minor and patch versions.

Many properties were added and removed from versions 3 to 4, which can be seen in the [release notes](https://docs.ted.europa.eu/EPO/latest/release-notes.html) but currently, there is no explicit declaration in the ontology. 

This is a discussion to have, and given the size and the rate of change of the Ontology, versioning through namespace could be evaluated as well.

I'm moving this into 'discussions'

**Reply by AchillesDougalis** [reply]
Besides the deprecation issue, the reason for deleting  [:isWithinFrameworkAgreement](https://github.com/OP-TED/ePO/blob/98b154c9a78256014a37f5328caaf3e8f9a0361f/implementation/ePO/owl_ontology/ePO_owl_core.ttl#L5691) was that  the predicate epo:followsRulesSetBy exists. 
![temp](https://github.com/OP-TED/ePO/assets/143995053/169cbab9-078c-433f-a9c9-e2284c4f60ef)
